### PR TITLE
Hide Shorts from YouTube Subscriptions page

### PR DESCRIPTION
**Pull request by vimode**
Added a new filter to remove this newly added section of *Shorts* in between the subscription feed. 
I am not sure if its the new UI or just an experiment. 

Added screenshot(s) for clarity. 
Before adding the filter

After adding the filter to my list of uBlock filters.
After adding newfilter

Hope I updated the filter list properly and provided sufficient information for the PR. Let me know if any changes are required.

Thank you.

### CODE REVIEW

**Comment by vimode** [comment]
Is the contributors list to be updated at contributors.json and then added to the template page or is it just for maintainers?

**Comment by xvello** [comment]
Heya @vimode, thanks for your contribution!

- I opened [this PR](https://github.com/letsblockit/letsblockit/pull/507) to add you to the contributor list. Once it's merged, you should indeed add yourself as a template contributor with that PR
- Could you please check whether this rule correctly hides that section? It should be a little more selective:
```adblock
www.youtube.com##ytd-browse ytd-reel-shelf-renderer
```
- If it does not work, could you please copy the DOM for this new element?
  - right-click
  - find and select the `ytd-rich-section-renderer`, it should be highlighted in the browser
  - right-click and use "copy outer html", then paste it in a comment here

**Comment by vimode** [comment]
Hey @xvello 

Thank you for guiding me through this. 

`www.youtube.com##ytd-browse ytd-reel-shelf-renderer` does not hide the shorts from /subscriptions page.

I have attached the DOM output of the tag `ytd-rich-shelf-renderer`. 
I removed some repeated HTML to make it smaller, the removed HTML is just repeated grid items for each Short. I added a comment for it in the pasted content below. 
But if you want I can paste it in full if that helps. 

```html
<ytd-rich-section-renderer class="style-scope ytd-rich-grid-renderer"><!--css-build:shady--><!--css-build:shady--><div id="content" class="style-scope ytd-rich-section-renderer"><ytd-rich-shelf-renderer class="style-scope ytd-rich-section-renderer" is-shorts="" style="--ytd-rich-shelf-items-count: 12;"><!--css-build:shady--><!--css-build:shady--><div id="dismissible" class="style-scope ytd-rich-shelf-renderer">
  <div id="rich-shelf-header" class="style-scope ytd-rich-shelf-renderer">
    <h2 class="style-scope ytd-rich-shelf-renderer">
      
        <yt-icon id="icon" class="style-scope ytd-rich-shelf-renderer"><!--css-build:shady--><!--css-build:shady--><yt-icon-shape class="style-scope yt-icon"><icon-shape class="yt-spec-icon-shape"><div style="width: 100%; height: 100%; fill: currentcolor;"><svg width="24" height="24" viewBox="0 0 24 24" focusable="false" style="pointer-events: none; display: block; width: 100%; height: 100%;">
  <g>
    <path d="M17.77,10.32l-1.2-.5L18,9.06a3.74,3.74,0,0,0-3.5-6.62L6,6.94a3.74,3.74,0,0,0,.23,6.74l1.2.49L6,14.93a3.75,3.75,0,0,0,3.5,6.63l8.5-4.5a3.74,3.74,0,0,0-.23-6.74Z" fill="red"></path>
    <polygon points="10 14.65 15 12 10 9.35 10 14.65" fill="#fff"></polygon>
  </g>
</svg></div></icon-shape></yt-icon-shape></yt-icon>
        <yt-img-shadow id="avatar" class="style-scope ytd-rich-shelf-renderer" disable-upgrade="" hidden="">
        </yt-img-shadow>
        <div id="title-container" class="style-scope ytd-rich-shelf-renderer">
          <div id="title-text" class="style-scope ytd-rich-shelf-renderer">
            <span id="title" class="style-scope ytd-rich-shelf-renderer">Shorts</span>
            <ytd-badge-supported-renderer id="featured-badge" class="style-scope ytd-rich-shelf-renderer" disable-upgrade="" hidden="">
            </ytd-badge-supported-renderer>
          </div>
          <div id="subtitle-text" class="style-scope ytd-rich-shelf-renderer">
            <ytd-badge-supported-renderer id="paygated-featured-badge" class="style-scope ytd-rich-shelf-renderer" disable-upgrade="" hidden="">
            </ytd-badge-supported-renderer>
            <yt-formatted-string id="subtitle" class="style-scope ytd-rich-shelf-renderer" is-empty=""><!--css-build:shady--><!--css-build:shady--><yt-attributed-string class="style-scope yt-formatted-string"></yt-attributed-string></yt-formatted-string>
          </div>
        </div>
      <dom-if class="style-scope ytd-rich-shelf-renderer"><template is="dom-if"></template></dom-if>
      <dom-if class="style-scope ytd-rich-shelf-renderer"><template is="dom-if"></template></dom-if>
    </h2>
    <div id="menu" class="style-scope ytd-rich-shelf-renderer"><ytd-menu-renderer class="style-scope ytd-rich-shelf-renderer" safe-area="" condensed="" icon="yt-icons:more" has-items="1"><!--css-build:shady--><!--css-build:shady--><div id="top-level-buttons-computed" class="top-level-buttons style-scope ytd-menu-renderer"><ytd-button-renderer class="style-scope ytd-menu-renderer" button-renderer="" button-next=""><!--css-build:shady--><yt-button-shape><a class="yt-spec-button-shape-next yt-spec-button-shape-next--text yt-spec-button-shape-next--call-to-action yt-spec-button-shape-next--size-m " style="" aria-label="View all" href="/feed/subscriptions/shorts" rel="nofollow" target="" force-new-state="true"><div class="cbox yt-spec-button-shape-next__button-text-content"><span class="yt-core-attributed-string yt-core-attributed-string--white-space-no-wrap" role="text">View all</span></div><yt-touch-feedback-shape style="border-radius: inherit;"><div class="yt-spec-touch-feedback-shape yt-spec-touch-feedback-shape--touch-response" aria-hidden="true"><div class="yt-spec-touch-feedback-shape__stroke" style=""></div><div class="yt-spec-touch-feedback-shape__fill" style=""></div></div></yt-touch-feedback-shape></a></yt-button-shape><tp-yt-paper-tooltip fit-to-visible-bounds="" offset="8" disable-upgrade=""></tp-yt-paper-tooltip></ytd-button-renderer></div><div id="flexible-item-buttons" class="style-scope ytd-menu-renderer"></div><yt-icon-button id="button" class="dropdown-trigger style-scope ytd-menu-renderer" style-target="button" hidden=""><!--css-build:shady--><!--css-build:shady--><button id="button" class="style-scope yt-icon-button"><yt-icon class="style-scope ytd-menu-renderer"><!--css-build:shady--><!--css-build:shady--><yt-icon-shape class="style-scope yt-icon"><icon-shape class="yt-spec-icon-shape"><div style="width: 100%; height: 100%; fill: currentcolor;"><svg height="24" viewBox="0 0 24 24" width="24" focusable="false" style="pointer-events: none; display: block; width: 100%; height: 100%;"><path d="M7.5 12c0 .83-.67 1.5-1.5 1.5s-1.5-.67-1.5-1.5.67-1.5 1.5-1.5 1.5.67 1.5 1.5zm4.5-1.5c-.83 0-1.5.67-1.5 1.5s.67 1.5 1.5 1.5 1.5-.67 1.5-1.5-.67-1.5-1.5-1.5zm6 0c-.83 0-1.5.67-1.5 1.5s.67 1.5 1.5 1.5 1.5-.67 1.5-1.5-.67-1.5-1.5-1.5z"></path></svg></div></icon-shape></yt-icon-shape></yt-icon></button><yt-interaction id="interaction" class="circular style-scope yt-icon-button"><!--css-build:shady--><!--css-build:shady--><div class="stroke style-scope yt-interaction"></div><div class="fill style-scope yt-interaction"></div></yt-interaction></yt-icon-button><yt-button-shape id="button-shape" version="modern" class="style-scope ytd-menu-renderer" disable-upgrade="" hidden=""></yt-button-shape></ytd-menu-renderer></div>
  </div>
  <div id="contents" class="style-scope ytd-rich-shelf-renderer"><ytd-rich-item-renderer class="style-scope ytd-rich-shelf-renderer" items-per-row="3" is-shelf-item="" is-slim-media="" is-reel-item-style-avatar-circle=""><!--css-build:shady--><!--css-build:shady--><div id="content" class="style-scope ytd-rich-item-renderer"><ytd-rich-grid-slim-media class="style-scope ytd-rich-item-renderer" remove-max-width="" lockup="true" is-short=""><!--css-build:shady--><!--css-build:shady--><div id="dismissible" class="style-scope ytd-rich-grid-slim-media">
  <ytd-thumbnail rich-grid-thumbnail="" use-hovered-property="" width="9999" class="style-scope ytd-rich-grid-slim-media" size="large" is-original-aspect-ratio="" computed-object-fit="CONTAIN" loaded=""><!--css-build:shady--><!--css-build:shady--><a id="thumbnail" class="yt-simple-endpoint inline-block style-scope ytd-thumbnail" aria-hidden="true" tabindex="-1" rel="null" href="/shorts/Xvo7vjmEoR0">
  <yt-image alt="" ftl-eligible="" notify-on-loaded="" notify-on-unloaded="" class="style-scope ytd-thumbnail"><img alt="" style="background-color: transparent;" class="yt-core-image--fill-parent-height yt-core-image--fill-parent-width yt-core-image yt-core-image--content-mode-scale-aspect-fit yt-core-image--loaded" src="https://i.ytimg.com/vi/Xvo7vjmEoR0/oar2.jpg?sqp=-oaymwEaCJUDENAFSFXyq4qpAwwIARUAAIhCcAHAAQY=&amp;rs=AOn4CLBk4A2dwH7lkYdE6LPYcF-kEKlGFg"></yt-image>
  
  <div id="overlays" class="style-scope ytd-thumbnail"></div>
  <div id="mouseover-overlay" class="style-scope ytd-thumbnail"></div>
  <div id="hover-overlays" class="style-scope ytd-thumbnail"></div>
</a>
</ytd-thumbnail>
  <div id="details" class="style-scope ytd-rich-grid-slim-media">
    <h3 class="style-scope ytd-rich-grid-slim-media">
      <a class="yt-simple-endpoint focus-on-expand style-scope ytd-rich-grid-slim-media" href="/shorts/Xvo7vjmEoR0" aria-label="Four GOLDEN dives 🥇 – 16 seconds – play video" title="Four GOLDEN dives 🥇">
        <ytd-badge-supported-renderer class="style-scope ytd-rich-grid-slim-media" disable-upgrade="" hidden="">
        </ytd-badge-supported-renderer>
        <span id="video-title" class="style-scope ytd-rich-grid-slim-media">Four GOLDEN dives 🥇</span>
      </a>
    </h3>
    <span id="metadata" class="style-scope ytd-rich-grid-slim-media" hidden="">
      
    </span>
    <ytd-video-meta-block class="grid style-scope ytd-rich-grid-slim-media byline-separated" rich-meta="" amsterdam-post-mvp=""><!--css-build:shady--><!--css-build:shady-->
<div id="metadata" class="style-scope ytd-video-meta-block">
  <div id="byline-container" class="style-scope ytd-video-meta-block" hidden="">
    <ytd-channel-name id="channel-name" class=" style-scope ytd-video-meta-block style-scope ytd-video-meta-block"><!--css-build:shady--><!--css-build:shady--><div id="container" class="style-scope ytd-channel-name">
  <div id="text-container" class="style-scope ytd-channel-name">
    <yt-formatted-string id="text" link-inherit-color="" title="" class="style-scope ytd-channel-name" is-empty="" ellipsis-truncate="" ellipsis-truncate-styling=""><!--css-build:shady--><!--css-build:shady--><yt-attributed-string class="style-scope yt-formatted-string"></yt-attributed-string></yt-formatted-string>
  </div>
  <tp-yt-paper-tooltip fit-to-visible-bounds="" class="style-scope ytd-channel-name" role="tooltip" tabindex="-1"><!--css-build:shady--><div id="tooltip" class="hidden style-scope tp-yt-paper-tooltip" style-target="tooltip">
  
    
  
</div>
</tp-yt-paper-tooltip>
</div>
<ytd-badge-supported-renderer class="style-scope ytd-channel-name" disable-upgrade="" hidden="">
</ytd-badge-supported-renderer>
</ytd-channel-name>
    <div id="separator" class="style-scope ytd-video-meta-block">•</div>
    <yt-formatted-string id="video-info" class="style-scope ytd-video-meta-block" is-empty="" hidden=""><!--css-build:shady--><!--css-build:shady--><yt-attributed-string class="style-scope yt-formatted-string"></yt-attributed-string></yt-formatted-string>
  </div>
  <div id="metadata-line" class="style-scope ytd-video-meta-block">
    
    <ytd-badge-supported-renderer class="inline-metadata-badge style-scope ytd-video-meta-block" system-icons="" hidden=""><!--css-build:shady--><!--css-build:shady--><dom-repeat id="repeat" as="badge" class="style-scope ytd-badge-supported-renderer"><template is="dom-repeat"></template></dom-repeat></ytd-badge-supported-renderer>
    <div id="separator" class="style-scope ytd-video-meta-block" hidden="">•</div>
    
      <span class="inline-metadata-item style-scope ytd-video-meta-block">630 views</span>
    <dom-repeat strip-whitespace="" class="style-scope ytd-video-meta-block"><template is="dom-repeat"></template></dom-repeat>
  </div>
</div>
<div id="additional-metadata-line" class="style-scope ytd-video-meta-block">
  <dom-repeat class="style-scope ytd-video-meta-block"><template is="dom-repeat"></template></dom-repeat>
</div>

</ytd-video-meta-block>
    <div id="menu" class="style-scope ytd-rich-grid-slim-media"><ytd-menu-renderer class="style-scope ytd-rich-grid-slim-media" safe-area=""><!--css-build:shady--><!--css-build:shady--><div id="top-level-buttons-computed" class="top-level-buttons style-scope ytd-menu-renderer"></div><div id="flexible-item-buttons" class="style-scope ytd-menu-renderer"></div><yt-icon-button id="button" class="dropdown-trigger style-scope ytd-menu-renderer" style-target="button"><!--css-build:shady--><!--css-build:shady--><button id="button" class="style-scope yt-icon-button" aria-label="More actions"><yt-icon class="style-scope ytd-menu-renderer"><!--css-build:shady--><!--css-build:shady--><yt-icon-shape class="style-scope yt-icon"><icon-shape class="yt-spec-icon-shape"><div style="width: 100%; height: 100%; fill: currentcolor;"><svg enable-background="new 0 0 24 24" height="24" viewBox="0 0 24 24" width="24" focusable="false" style="pointer-events: none; display: block; width: 100%; height: 100%;"><path d="M12 16.5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5-1.5-.67-1.5-1.5.67-1.5 1.5-1.5zM10.5 12c0 .83.67 1.5 1.5 1.5s1.5-.67 1.5-1.5-.67-1.5-1.5-1.5-1.5.67-1.5 1.5zm0-6c0 .83.67 1.5 1.5 1.5s1.5-.67 1.5-1.5-.67-1.5-1.5-1.5-1.5.67-1.5 1.5z"></path></svg></div></icon-shape></yt-icon-shape></yt-icon></button><yt-interaction id="interaction" class="circular style-scope yt-icon-button"><!--css-build:shady--><!--css-build:shady--><div class="stroke style-scope yt-interaction"></div><div class="fill style-scope yt-interaction"></div></yt-interaction></yt-icon-button><yt-button-shape id="button-shape" version="modern" class="style-scope ytd-menu-renderer" disable-upgrade="" hidden=""></yt-button-shape></ytd-menu-renderer></div>
  </div>
  
  <dom-if class="style-scope ytd-rich-grid-slim-media"><template is="dom-if"></template></dom-if>
  
    <ytd-badge-supported-renderer class="video-badge style-scope ytd-rich-grid-slim-media" disable-upgrade="">
    </ytd-badge-supported-renderer>
  <dom-if class="style-scope ytd-rich-grid-slim-media"><template is="dom-if"></template></dom-if>
</div>
<div id="dismissed" class="style-scope ytd-rich-grid-slim-media">
  <div id="dismissed-content" class="style-scope ytd-rich-grid-slim-media"></div>
</div>
<yt-interaction id="interaction" class="extended style-scope ytd-rich-grid-slim-media"><!--css-build:shady--><!--css-build:shady--><div class="stroke style-scope yt-interaction"></div><div class="fill style-scope yt-interaction"></div></yt-interaction>
</ytd-rich-grid-slim-media></div>
</ytd-rich-item-renderer>

<!-- <ytd-rich-item-renderer></ytd-rich-item-renderer>from above repeats a few times for each Short grid item -->

</div>

  <div class="button-container style-scope ytd-rich-shelf-renderer">
    <ytd-button-renderer class="expand-collapse-button style-scope ytd-rich-shelf-renderer" button-renderer="" button-next=""><!--css-build:shady--><yt-button-shape><button class="yt-spec-button-shape-next yt-spec-button-shape-next--text yt-spec-button-shape-next--mono yt-spec-button-shape-next--size-m yt-spec-button-shape-next--icon-only-default yt-spec-button-shape-next--workaround-icon-no-border-radius " style="" aria-label="Show more"><div class="yt-spec-button-shape-next__icon" aria-hidden="true"><yt-icon style="width: 24px; height: 24px;"><!--css-build:shady--><!--css-build:shady--><yt-icon-shape class="style-scope yt-icon"><icon-shape class="yt-spec-icon-shape"><div style="width: 100%; height: 100%; fill: currentcolor;"><svg height="24" viewBox="0 0 24 24" width="24" focusable="false" style="pointer-events: none; display: block; width: 100%; height: 100%;"><path d="m18 9.28-6.35 6.35-6.37-6.35.72-.71 5.64 5.65 5.65-5.65z"></path></svg></div></icon-shape></yt-icon-shape></yt-icon></div><yt-touch-feedback-shape style="border-radius: inherit;"><div class="yt-spec-touch-feedback-shape yt-spec-touch-feedback-shape--touch-response" aria-hidden="true"><div class="yt-spec-touch-feedback-shape__stroke" style=""></div><div class="yt-spec-touch-feedback-shape__fill" style=""></div></div></yt-touch-feedback-shape></button></yt-button-shape><tp-yt-paper-tooltip fit-to-visible-bounds="" offset="8" role="tooltip" tabindex="-1" style="inset: 1434.65px auto auto 755.383px;"><!--css-build:shady--><div id="tooltip" class="style-scope tp-yt-paper-tooltip hidden" style-target="tooltip">
  Show more
</div>
</tp-yt-paper-tooltip></ytd-button-renderer>
  </div>
  <div class="button-container style-scope ytd-rich-shelf-renderer" hidden="">
    <ytd-button-renderer class="expand-collapse-button style-scope ytd-rich-shelf-renderer" button-renderer="" button-next=""><!--css-build:shady--><yt-button-shape></yt-button-shape><tp-yt-paper-tooltip fit-to-visible-bounds="" offset="8" disable-upgrade=""> </tp-yt-paper-tooltip></ytd-button-renderer>
  </div>
</div>
<div id="dismissed" class="style-scope ytd-rich-shelf-renderer"></div>
</ytd-rich-shelf-renderer></div>
</ytd-rich-section-renderer>
```

**Comment by vimode** [comment]
If it helps, this is how I ended up with that filter using uBlock's cosmetic filtering tool.

yt-july-2023-shorts-3

**Review by JohnyP36 — APPROVED** [review]

**Comment by xvello** [comment]
Thanks for the DOM dump! Could you please try out the following rule?

```adblock
www.youtube.com##ytd-rich-shelf-renderer[is-shorts]
```

These might pop up on other pages, so I'm not scoping it for the submission page.
@JohnyP36 we also have several rules matching `ytd-reel-shelf-renderer` on several pages (sidebar, search, channel pages...), what do you say we replace them with a single `www.youtube.com##ytd-reel-shelf-renderer` rule to catch more future occurences?

**Comment by vimode** [comment]
`www.youtube.com##ytd-rich-shelf-renderer[is-shorts]`

Yes, this works.

**Comment by JohnyP36** [comment]
> @JohnyP36 we also have several rules matching ytd-reel-shelf-renderer on several pages (sidebar, search, channel pages...), what do you say we replace them with a single www.youtube.com##ytd-reel-shelf-renderer rule to catch more future occurences?

If I enable this rule, the following will happen: 
 - It does not block the shorts section on the homepage and subscription page
 - It does indeed block all shorts section in the search results (e.g. `https://www.youtube.com/results?search_query=organ+playing`)
 - It hides the shorts section on the homepage of a channel (e.g. `https://www.youtube.com/@dariialytvishko/featured`)

As people who are using this template don't have the option to block only certain parts (so, only shorts on subscription page and not in search results) we can simplify the template to replace all rules matching `##ytd-reel-shelf-renderer` with your suggestion. 
Even more as for mobile YouTube there is already the rule `m.youtube.com##ytm-reel-shelf-renderer` in the template.

Also, I found out that the rule `www.youtube.com##ytd-browse[page-subtype="subscriptions"] .ytd-thumbnail[href^="/shorts/"]:upward(ytd-grid-video-renderer,ytd-rich-item-renderer)` should be blocking the shorts section in the subscription page. As it now only blocks the videos/shorts and not the section. That rule should be replaced with `www.youtube.com##ytd-rich-shelf-renderer[is-shorts]`

File: data/filters/templates/youtube-shorts.yaml
Comment (JohnyP36) [inline data/filters/templates/youtube-shorts.yaml:25]: This section should - in my eyes - be removed as the goal of this rule is the same as the newly added `Hides Shorts section on the Subscriptions page`.

File: data/filters/templates/youtube-shorts.yaml
Comment (JohnyP36) [inline data/filters/templates/youtube-shorts.yaml:28]: Should be replaced with `www.youtube.com##ytd-rich-shelf-renderer[is-shorts]` and the description (between the `{{ }}` ) should be something like 
```
Hides Shorts sections on Homepage shelf and on the Subscription page
```

**Review by JohnyP36 — CHANGES_REQUESTED** [review]

**Comment by xvello** [comment]
Heya @vimode, https://github.com/letsblockit/letsblockit/pull/510 is currently being deployed and addresses the section you pointed out. I'm gonna close this PR, but I appreciate your contribution nonetheless. Don't hesitate to suggest more improvements!

**Pull request closed without merging**
